range_iterator.py

```diff
@@ -0,0 +1,23 @@
+class MyRange:
+
+    def __init__(self, start, stop, step):
+        self.current_val = start
+        self.stop = stop
+        self.step = step
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        while self.current_val <= (self.stop - self.step):
```

Comment:
And what about `self.current_val > (self.stop - self.step)`?

Reply:
In this case, we should stop adding steps (it would exceed the stop point), so I only add step if it's < or =

Reply:
Yes, but this is not something we would expect from the iterator.